seqr/views/apis/individual_api.py

```diff
@@ -806,6 +815,170 @@ def save_individuals_metadata_table_handler(request, project_guid, upload_file_i
+            family__project=project, participant_id__in=individuals_by_participant,
+        ).values('participant_id', 'guid', 'family_id')
+    }
+
+    family_variant_data = {}
+    finding_id_map = {}
+    genes = set()
+    for row in _iter_metadata_table(
+        metadata_files_path, FINDINGS_TABLE, request.user,
+            lambda r: r['participant_id'] in participant_individual_map and r['variant_type'] == 'SNV/INDEL',
+    ):
+        individual = participant_individual_map[row['participant_id']]
+        variant_id = '-'.join([row[col] for col in ['chrom', 'pos', 'ref', 'alt']])
+        key = (individual['family_id'], variant_id)
+        variant = {k: v for k, v in row.items() if v and v != 'NA'}
+        variant.update({
+            'pos': int(variant['pos']),
+            'genomeVersion': variant['variant_reference_assembly'].replace('GRCh', ''),
+            'transcripts': {},
+            'transcript': {
+                config.get('seqr_field', k): variant.pop(k, None) for k, config in TRANSCRIPT_FIELDS.items()
+            },
+            'genotypes': {individual['guid']: {'numAlt': 2 if variant['zygosity'] == 'Homozygous' else 1}},
+            'support_vars': [],
+        })
+        family_variant_data[key] = variant
+        genes.add(variant['gene'])
+        finding_id_map[variant['genetic_findings_id']] = variant_id
+
+    gene_symbols_to_ids = {k: v[0] for k, v in get_gene_ids_for_gene_symbols(genes).items()}
+    missing_genes = set()
+    for variant in family_variant_data.values():
+        gene = variant['gene']
+        transcript = variant.pop('transcript')
+        if gene in gene_symbols_to_ids:
+            variant.update({
+                'transcripts': {gene_symbols_to_ids[gene]: [transcript]},
+                'mainTranscriptId': transcript['transcriptId'],
+            })
+        else:
+            missing_genes.add(gene)
+        if variant.get('linked_variant') in finding_id_map:
+            variant['support_vars'].append(finding_id_map[variant['linked_variant']])
+
+    if missing_genes:
+        warnings.append(f'The following unknown genes were omitted in the findings tags: {", ".join(sorted(missing_genes))}')
+
+    num_new, num_updated = bulk_create_tagged_variants(
+        family_variant_data, tag_name=GREGOR_FINDING_TAG_TYPE, user=request.user, project=project,
+        get_metadata=lambda v: {k: v[k] for k in FINDING_METADATA_COLUMNS if k in v}
+    )
+    info.append(f'Loaded {num_new} new and {num_updated} updated findings tags')
+
+    response_json['projectsByGuid'] = {project_guid: {}}
+    response_json['familyTagTypeCounts'] = add_project_tag_types(response_json['projectsByGuid'], add_counts=True)
+
+    response_json['importStats'] = {'gregorMetadata': {'info': info, 'warnings': warnings}}
+    return create_json_response(response_json)
+
+
+def _iter_metadata_table(file_path, table_name, user, filter_row):
+    file_name = f'{file_path}/{table_name}.tsv'
```

Comment:
Is file the same as the one generated by the gregor report page in seqr?

Reply:
Kind of. All the centers submit the same reports and then the DCC makes them into one big report. But in general, we want the schema for the report we generate to match the schema for the metadata we import here